app/models/contract.rb

```diff
@@ -2,6 +2,10 @@
 class Contract < ApplicationRecord
   validates :start_date, presence: true
   validates :finish_date, presence: true
-  belongs_to :space
+  has_many :contract_spaces
```

Comment:
`has_and_belongs_to_many :spaces`

http://guides.rubyonrails.org/association_basics.html#the-has-many-association
2.6 The has_and_belongs_to_many Association
A has_and_belongs_to_many association creates a direct many-to-many connection with another model, with no intervening model. For example, if your application includes assemblies and parts, with each assembly having many parts and each part appearing in many assemblies, you could declare the models this way: